app/Services/MetaData.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Services;
+
+use DOMDocument;
+use Exception;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Cache;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Str;
+
+final readonly class MetaData
+{
+    /**
+     * The Open Graph data.
+     *
+     * @var Collection<string, string>
+     */
+    private Collection $data;
+
+    /**
+     * Fetch the Open Graph data for a given URL.
+     */
+    public function __construct(private string $url)
+    {
+        /** @var Collection<string, string> $cachedData */
+        $cachedData = Cache::remember(
+            Str::of($url)->slug()->prepend('preview_')->value(),
+            now()->addYear(),
+            fn (): Collection => $this->getData()
+        );
+
+        $this->data = $cachedData;
+    }
+
+    /**
+     * Fetch the parsed meta-data for a given URL.
+     *
+     * @return Collection<string, string>
+     */
+    public static function fetch(string $url): Collection
+    {
+        return (new self($url))->data;
+    }
+
+    /**
+     * Get the meta-data for a given URL.
+     *
+     * @return Collection<string, string>
+     */
+    public function getData(): Collection
+    {
+        $data = collect();
+
+        try {
+            $response = Http::get($this->url);
```

Comment:
What about Http::head 🤔

Reply:
Interesting though, but we need the <head> tag from within the response body, Http::head returns the request headers from the url.